File: src/main/java/shop/ipwebshop/models/dto/Product.java
package shop.ipwebshop.models.dto;

import lombok.Data;
import java.math.BigDecimal;
import java.sql.Timestamp;


@Data
public class Product {
    private Integer id;
    private String title;
    private String description;
    private BigDecimal price;
    private String state;
    private Boolean sold;
    private Boolean deleted;
    private Timestamp postedTime;
    private String categoryName;
    private Integer userId;
    private String userUsername;

}

File: frontend/ip-webshop/src/app/models/product.ts
export class Product {
    id: number | null;
    title: string | null;
    description: string | null;
    price: number | null; 
    state: string | null;
    postedTime: Date | null;
    userId: number | null;
    userUsername: string | null; 
    userAvatarUrl: string | null;
    userLocation: string | null;
    constructor(
      id?: number, 
      title?: string, 
      description?: string, 
      price?: number, 
      state?: string, 
      postedTime?: Date,
      userId?: number,
      userUsername?: string,
      userAvatarUrl?: string,
      userLocation?: string
    ) {
      this.id = id || null;
      this.title = title || null;
      this.description = description || null;
      this.price = price || null;
      this.state = state || null;
      this.postedTime = postedTime || null;
      this.userId = userId || null;
      this.userUsername = userUsername || null;
      this.userAvatarUrl = userAvatarUrl || null;
      this.userLocation = userLocation || null;
    }
  }